### DISCUSSION TITLE
Unknown devices (MX400?)

### DISCUSSION THREAD

**Discussion by Elektron79** [emsesp/EMS-ESP32#2214] [General]
![Uploading IMG_0870.jpeg…]()
### DESCRIPTION

I have a Buderus WLW186i Heatpump with the „new“ MX400 Module which is the successor of the well known MX300 device.
In the Device Overview I have two „unknown“ devices. I expect that at least one is the MX400.
How can I support to enable the project to support also the MX400 Module?

### REQUESTED INFORMATION

_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ x] Searched the issue in [issues](https://github.com/emsesp/EMS-ESP32/issues)
- [ ] Searched the issue in [discussions](https://github.com/emsesp/EMS-ESP32/discussions)
- [ ] Searched the issue in the [docs](https://emsesp.org/Troubleshooting/)
- [ ] Searched the issue in the [chat](https://discord.gg/3J3GgnzpyT)
- [ ] Provide the System information in the area below, taken from `http://<IP>/api/system`

```json


{"system":{"version":"3.7.1-dev.5","uptime":"000+00:07:04.950","uptimeSec":424,"platform":"ESP32S3","cpuType":"ESP32-S3","arduino":"ESP32 Arduino v2.0.17","sdk":"v4.4.7-dirty","freeMem":197,"maxAlloc":183,"freeCaps":8326,"usedApp":2284,"freeApp":3412,"partition":"app0","resetReason":"Software reset CPU / Software reset CPU","psram":true,"psramSize":8189,"freePsram":8130,"model":"","temperature":39.8},"network":{"network":"WiFi","hostname":"ems-esp","RSSI":-42,"BSSID":"set","TxPowerSetting":0,"staticIP":false,"lowBandwidth":false,"disableSleep":true,"enableMDNS":true,"enableCORS":false,"APProvisionMode":"disconnected","APSecurity":"wpa2","APSSID":"ems-esp"},"ntp":{"NTPStatus":"connected","enabled":true,"server":"time.google.com","tzLabel":"Europe/Amsterdam"},"mqtt":{"MQTTStatus":"connected","MQTTPublishes":30,"MQTTQueued":0,"MQTTPublishFails":0,"MQTTConnects":1,"enabled":true,"clientID":"esp32-8abd4d74","keepAlive":60,"cleanSession":false,"entityFormat":0,"base":"ems-esp","discoveryPrefix":"homeassistant","discoveryType":0,"nestedFormat":1,"haEnabled":false,"mqttQos":0,"mqttRetain":false,"publishTimeHeartbeat":60,"publishTimeBoiler":60,"publishTimeThermostat":60,"publishTimeSolar":60,"publishTimeMixer":60,"publishTimeWater":10,"publishTimeOther":60,"publishTimeSensor":60,"publishSingle":false,"publish2command":false,"sendResponse":false},"syslog":{"enabled":false},"sensor":{"temperatureSensors":0,"temperatureSensorReads":0,"temperatureSensorFails":0,"analogSensors":0,"analogSensorReads":0,"analogSensorFails":0},"api":{"APICalls":0,"APIFails":0},"bus":{"busStatus":"connected","busProtocol":"Buderus","busTelegramsReceived":1895,"busReads":532,"busWrites":0,"busIncompleteTelegrams":2,"busReadsFailed":0,"busWritesFailed":0,"busRxLineQuality":100,"busTxLineQuality":100},"settings":{"boardProfile":"S32S3","locale":"en","txMode":1,"emsBusID":11,"showerTimer":false,"showerMinDuration":180,"showerAlert":false,"hideLed":false,"noTokenApi":false,"readonlyMode":false,"fahrenheit":false,"dallasParasite":false,"boolFormat":1,"boolDashboard":1,"enumFormat":1,"analogEnabled":true,"telnetEnabled":true,"maxWebLogBuffer":50,"webLogBuffer":22,"modbusEnabled":false,"forceHeatingOff":false,"developerMode":false},"devices":[{"type":"boiler","name":"XCU_THH/CS*800i, Logatherm WLW*","deviceID":"0x08","productID":8,"brand":"Buderus","version":"00.00","entities":162,"handlersReceived":"0xBF 0xD1 0xE3 0xE4 0xE5 0xE9 0x0494 0x0495 0x048F","handlersFetched":"0x14 0xE6 0xEA 0x048D 0x04A2 0x0485 0x0486 0x0492 0x0488 0x0484 0x048B 0x0491 0x0499 0x049C 0x049D 0x04AE 0x04AF 0x02CC 0x04AA 0x04A7 0x02D6","handlersPending":"0x10 0x11 0xC2 0x15 0x1C 0x18 0x19 0x1A 0x35 0x34 0x2A","handlersIgnored":"0x023E 0x0240 0x0291 0x0292 0x0293 0x0294 0x029B 0x02A5 0x02F5 0x04A8 0x04A9 0x04AB 0x04AC 0x04CA 0x04CB 0x02E0 0x02EA 0x061E 0x04D8"},{"type":"thermostat","name":"HMI800.2/Rego 3000, UI800, Logamatic BC400","deviceID":"0x10","productID":253,"brand":"","version":"47.10","entities":60,"handlersReceived":"0x06 0x02BA 0x02BB 0x02BC 0x0267","handlersFetched":"0x02A5 0x02B9 0x02AF 0x029B 0x0471 0x02CC 0x0467 0x0291 0x0292 0x0293 0x0294 0x02F5 0x023A 0x0240 0xBB 0x023E","handlersPending":"0xA3 0xA2 0x12 0x13 0x02A6 0x02B0 0x029C 0x0472 0x02A7 0x02B1 0x029D 0x0473 0x02A8 0x02B2 0x029E 0x0474 0x02A9 0x02BD 0x02B3 0x029F 0x0475 0x02AA 0x02BE 0x02B4 0x02A0 0x0476 0x02AB 0x02BF 0x02B5 0x02A1 0x0477 0x02AC 0x02C0 0x02B6 0x02A2 0x0478 0x02CE 0x0468 0x02D0 0x0469 0x02D2 0x046A 0x031D 0x031E","handlersIgnored":"0x0507 0x0508 0x0509 0x050A 0x059D 0xBF"},{"type":"generic","name":"MX400/unknown","deviceID":"0x48","productID":17,"brand":"","version":"10.00","entities":0,"handlersIgnored":"0x04D6 0x04D7 0x0495 0x0494 0x04AF 0x04AE 0xF9 0xBF"},{"type":"generic","name":"MX400/unknown","deviceID":"0x4B","productID":17,"brand":"","version":"10.00","entities":0}]}


```

### TO REPRODUCE

connect a mx400 device to your system.

### EXPECTED BEHAVIOUR

MX400 is discovered

### SCREENSHOTS

_If applicable, add screenshots to help explain your issu

### ADDITIONAL CONTEXT

_Add any other context about the issue here._

**(Please remember to close the issue when it has been addressed)**

**Comment by MichaelDvP** [comment]
> "type":"generic","name":"MX400/unknown","deviceID":"0x4B","productID":17,

Yes both new devices are the MX400 (@proddy : nice to have the telegram 01 name here), don't know  whyit uses two device ids.
I'll add to database,so it will show up as gateway. But gateways does not have any own entiy, so you miss nothing meanwhile.

**Comment by schreierc84** [comment]
Hi folks,
for me the MX400 shows up actually three times:
![image](https://github.com/user-attachments/assets/7e819b02-1e2e-4361-9f6b-3802177a0d24)

Please find my system info below
[emsesp_system_info.json](https://github.com/user-attachments/files/18920954/emsesp_system_info.json)

Hope this can help to analyse.

Br
Christian

**Reply by MichaelDvP** [reply]
Yes, the MX400 uses 3 device-Ids: 0x48, 0x4B and 0x50.

**Comment by tubicenNEW** [comment]
Hi, I have the exact same setup — Buderus Logatherm **WLW186i-12** with **BC400** and the **MX400** communication module.

I'm commenting because of a related issue that might be relevant for others with this hardware: **the heat pump only switches into cooling mode when the EMS-ESP gateway is NOT connected to the EMS bus.** With the gateway plugged into the service jack, cooling never engages. As soon as I disconnect it, the unit goes into cooling normally.

Important: this wasn't my own guess. My heating installer diagnosed it, and the Buderus service technician confirmed it — both identified the third-party EMS gateway on the bus as the reason the cooling doesn't engage.

My EMS-ESP is currently on the default **Service Key (0x0B)** Bus ID. Since the MX400 already occupies 0x48/0x4B/0x50, I'm planning to move EMS-ESP to **Gateway 2 (0x49)** to get it off the reserved service-key slot, and then check whether cooling works with the gateway reconnected.

Questions for anyone running the same WLW186i + MX400 combo:
- Does your heat pump cool normally **with** EMS-ESP connected to the bus?
- If you had this problem, did changing the Bus ID away from 0x0B (e.g. to 0x49) fix it — or did it turn out to be a Tx-timing / bus-load issue that the ID change didn't solve?

I can't attach a trace log or system info at the moment, because I'm deliberately keeping the gateway disconnected from the heat pump so it keeps cooling during the next days. Just to be sure it really does. I'll follow up with logs (a trace during a cooling request, with and without the gateway) once I reconnect to test the 0x49 change.

Thanks!

**Reply by bbqkees** [reply]
The solution to this specific problem is to set the EMS Tx mode to EMS+
